### ISSUE TITLE
Meta::getTables should list also PARTITIONED TABLE

### ISSUE THREAD

**Issue by MirkoTeran** [jOOQ/jOOQ#14368]
### Your question

I'm having some issues listing tables from PG database. Tables that are partitioned are not listed, only the individual partitions are. Is there a way to do this trough jOOQ?

I've seen that MetaImpl calls getTables using {"FOREIGN TABLE", "MATERIALIZED VIEW", "SYSTEM_TABLE", "SYSTEM_VIEW", "TABLE", "VIEW"} types, skipping the "PARTITIONED TABLE" I would like. But I can't find a way to override this. Is this part of issues like 7844 and will be added later?

### jOOQ Version

3.17.4

### Database product and version

PostgreSQL 11.2, compiled by Visual C++ build 1914, 64-bit

### Java Version

OpenJDK 17

### OS Version

Microsoft Windows [Version 10.0.19044.2251]

### JDBC driver name and version (include name if unofficial driver)

org.postgresql.Driver version 42.5.1

**Comment by lukaseder** [comment]
Thanks a lot for your report. That's definitely not something we want to exclude, there. `DatabaseMetaData::getTableTypes` currently lists this on PostgreSQL 15:

```
+------------------+
|TABLE_TYPE        |
+------------------+
|FOREIGN TABLE     |
|INDEX             |
|MATERIALIZED VIEW |
|PARTITIONED INDEX |
|PARTITIONED TABLE |
|SEQUENCE          |
|SYSTEM INDEX      |
|SYSTEM TABLE      |
|SYSTEM TOAST INDEX|
|SYSTEM TOAST TABLE|
|SYSTEM VIEW       |
|TABLE             |
|TEMPORARY INDEX   |
|TEMPORARY SEQUENCE|
|TEMPORARY TABLE   |
|TEMPORARY VIEW    |
|TYPE              |
|VIEW              |
+------------------+
```

I think the reason why we explicitly list the table types was that (at some point), the JDBC driver wouldn't accept `null` as an argument (?) or it produced object types that we don't like, including indexes and sequences and types. Probably the latter.

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.18.0, 3.17.6 (https://github.com/jOOQ/jOOQ/issues/14369), and 3.16.12 (https://github.com/jOOQ/jOOQ/issues/14370)
